### DISCUSSION TITLE
How to install CLI

### DISCUSSION THREAD

**Discussion by tinca** [ultralytics/ultralytics#8014] [Q&A] [answered]
Hi,
I am aware that pip install is the recommended way to get access for ultralytics tools. However I do not want to use them as long as I really need them. The CLI would be the choice of tool for me at the beginning.
The docs talk about using CLI commands, but I am really feeling dumb, because I fail to find a description about how to install it.
I did a pip3 install ultralytics, but 'yolo' from the commad line gives:
-bash: yolo: command not found

Your help is appreciated!

**Comment by tinca** [comment]
I found this among the issues:
https://github.com/ultralytics/ultralytics/issues/6281
Although in my case /usr/local/bin is writable for my user it contains no yolo script.
I am on Mac, so it may be OS specific.

**Reply by tinca** [reply]
I tried to create the yolo script based on the above issue, but it does not work.
Please fix this for Mac users!
Until then, can I get a working yolo script version?

**Reply by glenn-jocher** [reply, marked as answer]
@tinca hello! I'm sorry to hear you're having trouble with the CLI installation. The `pip install ultralytics` command should install the necessary packages and provide you with access to the `yolo` command. However, it seems there might be an issue with the path configuration on your Mac.

To resolve this, you can try the following steps:

1. Ensure that the Python package installation path is included in your system's `PATH` environment variable. This is typically `$HOME/Library/Python/<version>/bin` or `/usr/local/bin` for a Mac.
2. Check if the `yolo` script is indeed in the aforementioned directories. If not, you might need to locate where the `yolo` executable was installed.
3. If you're unable to find the `yolo` script, you can manually add the directory containing the `yolo` script to your `PATH`.

If these steps don't resolve the issue, you might consider using the Python interface directly for now, which can be accessed by running Python scripts that import the `ultralytics` package and use its functionalities.

We appreciate your patience and are working to ensure better compatibility and ease of use across all platforms, including Mac. If you continue to experience difficulties, please provide us with additional details about your environment and installation process, and we'll do our best to assist you further. Thank you for bringing this to our attention! 🛠️

**Comment by tinca** [comment]
Hi Glenn,

I found it at $HOME/Library/Python/<version>/bin, added this to my path and it works now.
Thank you for your help.

**Reply by pderrenger** [reply]
@tinca hello! I'm glad to hear that you've found the solution and that the CLI is working for you now. If you have any more questions or need further assistance, feel free to reach out. Happy detecting with YOLOv8! 😊🚀

**Comment by hbarnard** [comment]
(sudo probably, I'm root on a temporary system) 
```
apt install pipx
pipx install ultralytics
pipx ensurepath
apt install libglu1-mesa-dev
``` (for missing libGL.so.1)

CLI is found now. The rest of the project is Perl, so command line access 'may' be simpler)

**Reply by glenn-jocher** [reply]
Thanks for sharing — `pipx install ultralytics` is a solid isolated option, and on headless systems the [`ultralytics-opencv-headless` package](https://docs.ultralytics.com/quickstart#headless-server-installation) can avoid `libGL.so.1` issues; for standard installs the `yolo` CLI is installed with `ultralytics`, and if it isn’t found after `pip install`, it’s usually a PATH or venv activation issue covered in the [Quickstart](https://docs.ultralytics.com/quickstart) and [CLI guide](https://docs.ultralytics.com/usage/cli).
